code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -1112,6 +1112,17 @@ var/global/list/damage_icon_parts = list() //see UpdateDamageIcon()
 			overlay.Blend(rgb(src.r_wing2, src.g_wing2, src.b_wing2), wing_style.color_blend_mode)
 			wing_s.Blend(overlay, ICON_OVERLAY)
 			qdel(overlay)
+		if(wing_style.extra_overlay2)
+			var/icon/overlay = new/icon("icon" = wing_style.icon, "icon_state" = wing_style.extra_overlay2)
+			if(wing_style.ani_state)

```

Comment:
Seems to be wasting a new icon operation here, which is bad news. Haven't pulled it down and tested, but I think this block could be simplified to:
````
var/icon/overlay = new/icon("icon" = wing_style.icon, "icon_state" = (wing_style.ani_state ? wing_style.extra_overlay2_w : wing_style.extra_overlay2))
overlay.Blend(rgb(src.r_wing3, src.g_wing3, src.b_wing3), wing_style.color_blend_mode)
wing_s.Blend(overlay, ICON_OVERLAY)
qdel(overlay)
````
Also not sure if you need to qdel `overlay` as it's an icon and should naturally fall out of scope and get GC'd.